src/main/java/com/j256/ormlite/dao/Dao.java

```diff
@@ -197,6 +198,23 @@
 	 */
 	public T createIfNotExists(T data) throws SQLException;
 
+	/**
+	 * This is a convenience method to creating a data item but only if the ID does not already exist in the table. This
+	 * extracts the id from the data parameter, does a {@link #queryForId(Object)} on it, returning the data if it
+	 * exists. If it does not exist {@link #create(Object)} will be called with the parameter.
+	 *
+	 * Unlike {@link #createIfNotExists(Object)}, this method able to initialize a new object lazily.
+	 *
+	 * <p>
+	 * <b>NOTE:</b> This method is synchronized because otherwise race conditions would be encountered if this is used
+	 * by multiple threads.
+	 * </p>
+	 *
+	 * @return Either the data parameter if it was inserted (now with the ID field set via the create method) or the
```

Comment:
This `@return` needs to be updated.  I think it was just copied from the above method.